altair/vegalite/v5/schema/channels.py

```diff
@@ -30945,7 +30945,7 @@ def encode(
         text: Optional[str | Text | Map | TextDatum | TextValue] = Undefined,
         theta: Optional[str | Theta | Map | ThetaDatum | ThetaValue] = Undefined,
         theta2: Optional[str | Theta2 | Map | Theta2Datum | Theta2Value] = Undefined,
-        tooltip: Optional[str | Tooltip | Map | list | TooltipValue] = Undefined,
+        tooltip: Optional[OneOrSeq[str | Tooltip | Map | TooltipValue]] = Undefined,
```

Comment:
- 3 extra characters
- doesn't break into a new line
- generic over `Sequence` instead of `list`

Personally think this reads nicely:

> tooltip is optional, accepting one or a `Sequence` of `str`, `Tooltip`, `Map`, `TooltipValue`

Reply:
What I have in mind for the `TypedDict`, is that since it doesn't need a default (`Undefined`) we can reduce this further there:

```py
tooltip_chart: Optional[OneOrSeq[str | Tooltip | Map | TooltipValue]] = Undefined
tooltip_tdict: OneOrSeq[str | Tooltip | Map | TooltipValue]
```

Passing `Undefined` doesn't make any sense from the context of `pl.DataFrame.plot`

Reply:
I like the usage of `OneOrSeq` and agree that `Undefined` is not relevant for these type hints.